Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/named_params_helper.h

```diff
@@ -56,6 +58,40 @@ class GetGeomTraits
   > ::type  type;
 };
 
+template<typename PolygonMesh, typename PropertyTag>
+class property_map_selector
+{
+public:
+  typedef typename boost::graph_has_property<PolygonMesh, PropertyTag>::type Has_internal_pmap;
+  typedef typename boost::mpl::if_c< Has_internal_pmap::value
+                          , typename boost::property_map<PolygonMesh, PropertyTag>::type
+                          , typename boost::cgal_no_property::type
+  >::type type;
+
+  type get_pmap(const PropertyTag& p, const PolygonMesh& pmesh)
+  {
+    return get_impl(p, pmesh, Has_internal_pmap());
+  }
+
+private:
+  type get_impl(const PropertyTag&, const PolygonMesh&, CGAL::Tag_false)
+  {
+    return type(); //boost::cgal_no_property::type
+  }
+  type get_impl(const PropertyTag& p, const PolygonMesh& pmesh, CGAL::Tag_true)
+  {
+    return get(p, pmesh);
+  }
+};
+
+template<typename PolygonMesh, typename PropertyTag>
+typename property_map_selector<PolygonMesh, PropertyTag>::type
+get_property_map(const PropertyTag& p, const PolygonMesh& pmesh)
+{
+  property_map_selector<PolygonMesh, PropertyTag> pms;
+  return pms.get_pmap(p, pmesh);
+}
+
```

Comment:
From what I understand, this removes the need of internal point property maps ? Shall we update the documentation of all these functions?